ecdsa.c

```diff
@@ -829,6 +849,30 @@ void ecdsa_get_pubkeyhash(const uint8_t *pub_key, uint8_t *pubkeyhash)
 	MEMSET_BZERO(h, sizeof(h));
 }
 
+#if USE_ETHEREUM
+int ecdsa_get_ethereum_pubkeyhash(const uint8_t *pub_key, uint8_t *pubkeyhash)
+{
+	uint8_t h[65];
+	SHA3_CTX ctx;
+
+	if (!ecdsa_uncompress_pubkey(&secp256k1, pub_key, h)) {
```

Comment:
True about `h`.

Actually if you look up the chain, I believe it is more the duty of `ecdsa_read_pubkey` to clean `pub` if `ecdsa_validate_pubkey` fails. The same issue is present in `ecdsa_verify_digest`. @prusnak, what do you think?